validator/sawtooth_validator/gossip/signature_verifier.py

```diff
@@ -47,15 +47,13 @@ def validate_block(block):
                            block.header_signature,
                            header.signer_pubkey)
 
+    if not valid:
+        LOGGER.debug("block failed signature validation: %s",
+                     block.header_signature)
+
```

Comment:
Why bother checking the batches if you have already discerned that the block header signature is invalid?  Why not:
```
if not signing.verify(...):
    LOGGER.debug(...)
    return False
```